.github/workflows/ci.yaml

```diff
@@ -322,6 +322,51 @@ jobs:
           set -euxo pipefail
           mv linux/rust.require.cmake linux/rust.cmake
           flutter test -d linux integration_test/main.dart --verbose
+  
+  dart_web_test:
+    name: Dart (Web) unit tests
+    runs-on: ubuntu-latest
+    
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dart-lang/setup-dart@v1
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          target: wasm32-unknown-unknown
+          components: rust-src,rustfmt
+      - name: Install ffigen
+        run: dart pub global activate ffigen
+      - name: Cache dependencies
+        id: test-cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            frb_codegen/target
+            frb_example/pure_dart/rust/target/
+            frb_example/pure_dart/dart/.local-chromium
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - name: Install crates
+        if: steps.test-cache.outputs.cache-hit != 'true'
+        run: |
+          sh -c "$(curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf)" '' -f
+      - name: Setup Dart dependencies
+        run: dart pub get
+        working-directory: frb_dart
+      - name: Run tests
+        working-directory: frb_example/pure_dart/dart
+        timeout-minutes: 20
+        run: dart pub get &&
+          dart run ../../../frb_dart/bin/serve.dart
+          -c ../rust --dart-input lib/main.web.dart --root web/ --run-tests
```

Comment:
Wait... This seems to have a big problem: Valgrind does not check web memory problems!

P.S. Also not ok if using `valgrind dart run serve.dart`, b/c that would check the serve.dart, not the generated wasm

Reply:
I don't think there are any suitable replacements yet for Valgrind on the web. In the long term, might I suggest using [`miri`](https://github.com/rust-lang/miri) instead?

Reply:
Good idea!

But then how to ensure no memory problem for this PR :/ There are quite a few unsafe operations in this PR, and we all know memory safety is very critical

Reply:
I think this PR is already quite long, so we should migrate to `miri` in a separate PR.

Reply:
Sure :) Maybe we can make an issue tracking it

Reply:
Created #641 to resolve this issue.